extension/observer/hostobserver/README.md

```diff
@@ -0,0 +1,29 @@
+# Host Observer Extension
+
+**Status: beta**
+
+Looks at the current host for listening network endpoints. It will look for all listening sockets on TCP and UDP over IPv4 and IPv6.
+
+It uses the /proc filesystem and requires the SYS_PTRACE and DAC_READ_SEARCH capabilities so that it can determine what processes own the listening sockets.
+
+### Configuration
+
+#### `refresh_interval`
+
+Determines how to look for changes in endpoints.
+
+default: `10s`
+
+### Endpoint Variables
+
+Endpoint variables exposed by this observer are as follows.
+
+| Variable  | Description                                                                                |
+|-----------|--------------------------------------------------------------------------------------------|
+| type.port | `true`                                                                                     |
+| name      | name of the process associated to the port                                                 |
+| port      | port number                                                                                |
+| command   | full command used to invoke this process, including the executable itself at the beginning |
+| is_ipv6   | `true` if the endpoint is IPv6                                                             |
+| protocol  | "TCP" or "UDP"                                                                             |
```

Comment:
This is really the transport protocol and IPv4 vs v6 is the network protocol.  That distinction was not made very clear in the Smart Agent host observer.

Reply:
Would you rather have endpoint variables called `transport_protocol` and `network_protocol` instead of `protocol` and `is_ipv6`?

Reply:
Discussed with @jrcamp as well and we were thinking of updating these variables to `transport_protocol` (either `TCP` or `UDP`) and `network_protocol` (either `ipv4` or `ipv6`). What do you think @keitwb ?

Reply:
ptal @jrcamp @keitwb - https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/473